pkg/controllermanager/controller/credentialsbinding/reconciler.go

```diff
@@ -78,25 +82,43 @@ func (r *Reconciler) reconcile(ctx context.Context, credentialsBinding *security
 	}
 	kind := credential.GetObjectKind().GroupVersionKind().Kind
 
-	if !controllerutil.ContainsFinalizer(credential, gardencorev1beta1.ExternalGardenerName) {
+	if !controllerutil.ContainsFinalizer(credential, finalizerName) {
 		log.Info("Adding finalizer", kind, client.ObjectKeyFromObject(credential)) //nolint:logcheck
-		if err := controllerutils.AddFinalizers(ctx, r.Client, credential, gardencorev1beta1.ExternalGardenerName); err != nil {
+		if err := controllerutils.AddFinalizers(ctx, r.Client, credential, finalizerName); err != nil {
 			return fmt.Errorf("could not add finalizer to %s: %w", kind, err)
 		}
 	}
 
+	// TODO(dimityrmirchev): Remove the ExternalGardenerName finalizer handling in a future release
+	if controllerutil.ContainsFinalizer(credential, gardencorev1beta1.ExternalGardenerName) {
+		log.Info("Removing finalizer", kind, client.ObjectKeyFromObject(credential)) //nolint:logcheck
```

Comment:
Now when the controller handles two finalizers, it would be nice to log the finalizer name.

```suggestion
		log.Info("Removing finalizer", "finalizer", gardencorev1beta1.ExternalGardenerName, kind, client.ObjectKeyFromObject(credential)) //nolint:logcheck
```